synapse/api/auth.py

```diff
@@ -383,19 +350,24 @@ async def _get_appservice_user_id_and_device_id(
                     Codes.EXCLUSIVE,
                 )
 
-        return effective_user_id, effective_device_id, app_service
+        return create_requester(
+            effective_user_id, app_service=app_service, device_id=effective_device_id
+        )
 
     async def get_user_by_access_token(
         self,
         token: str,
         allow_expired: bool = False,
-    ) -> TokenLookupResult:
+        mark_as_used: bool = False,
+    ) -> Requester:
         """Validate access token and get user_id from it
 
         Args:
             token: The access token to get the user by
             allow_expired: If False, raises an InvalidClientTokenError
                 if the token is expired
+            mark_as_used: Mark the token as used, if it was used to
+                authenticate a regular C-S API request
```

Comment:
isn't it more about the type of token than what it was used to authenticate (which this function doesn't know)?

```suggestion
            mark_as_used: Mark the token as used, if it was a regular
                C-S access token
```

Reply:
sure, getting rid of it seems sensible.

Reply:
I got rid of the parameter in 1670733e7